server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/next/ESNextDiskBBQVectorsWriter.java

```diff
@@ -387,8 +388,33 @@ public void writeCentroids(
         // TODO: sort centroids by global centroid (was doing so previously here)
         // TODO: sorting tanks recall possibly because centroids ordinals no longer are aligned
         if (centroidSupplier.size() > centroidsPerParentCluster * centroidsPerParentCluster) {
-            writeCentroidsWithParents(fieldInfo, centroidSupplier, globalCentroid, centroidOffsetAndLength, centroidOutput);
+
+            final CentroidGroups centroidGroups = buildCentroidGroups(fieldInfo, centroidSupplier);
+            {
+                // write vector ord -> centroid lookup table. We need to remap current centroid ordinals
+                // to the ordinals on the parent / child structure.
+                final int[] centroidOrdinalMap = new int[centroidSupplier.size()];
+                int idx = 0;
+                for (int[] centroidVectors : centroidGroups.vectors()) {
+                    for (int assignment : centroidVectors) {
+                        centroidOrdinalMap[assignment] = idx++;
+                    }
+                }
+                assert idx == centroidSupplier.size() : "Expected [" + centroidSupplier.size() + "], got [" + idx + "]";
+                // TODO: write ordinal to centroid map in a packed format?
+                for (int centroidAssignment : centroidAssignments) {
+                    centroidOutput.writeInt(centroidOrdinalMap[centroidAssignment]);
+                }
```

Comment:
ugh, I really wish we could shuffle things better so that the postings lists themselves get written in the new ordinal order, allowing for lists that have the same parent be near eachother. But, this is good for now.